straxen/plugins/led_calibration.py

```diff
@@ -0,0 +1,114 @@
+'''
+Dear nT analyser, 
+if you want to complain please contact: chiara@physik.uzh.ch, gvolta@physik.uzh.ch
+'''
+
+import strax
+import numpy as np
+import pandas as pd
+from tqdm import tqdm
+import numba
+from numba import njit
+
+#export, __all__ = strax.exporter()
+# QUESTION: what does 'export, __all__ = strax.exporter()' do?
+# ANSWER: [fill in]
+
+@strax.takes_config(
+    strax.Option('LED_window',
+                 default=(125, 250),
+                 help="Window (samples) where we expect the signal in LED calibration"),
+    strax.Option('noise_window',
+                 default=(350, 475),
+                 help="Window (samples) to analysis the noise"),
+    strax.Option('light_level',
+                 default=(0,248),
+                 help="Three different light level for XENON1T: (0,36), (37,126), (127,248). Defalt value: all the PMTs")
+)
+
+
+class LEDCalibration(strax.Plugin):
+    
+    '''
+    Preliminary version, several parameters to set during commisioning.
+    LEDCalibration returns: channel, time, dt, lenght, Area, amplitudeLED and amplitudeNOISE.
+    The new variables are:
+    - Area: Area computed in the given window, averaged over 6 windows that have the same starting sample and different end samples.
+    - amplitudeLED: peak amplitude of the LED on run in the given window.
+    - amplitudeNOISE: amplitude of the LED on run in a window far from the signal one.
+    '''
+    
+    __version__ = '0.0.8'
+    print(__version__)
+    depends_on = ('raw_records',)
+    # Options below copied from other plugins, need to be reviewed by an expert
+    data_kind = 'led_cal_0' 
+    compressor = 'zstd'
+    parallel = 'process'
+    rechunk_on_save = False
+    
+    dtype = [('area', np.int32, 'Area averaged in integration windows'),
+             ('amplitudeLED', np.int32, 'Amplitude in LED window'),
+             ('amplitudeNOISE', np.int32, 'Amplitude in off LED window'),
+             ('channel', np.int16, 'Channel'),
+             ('time', np.int64, 'Start time of the interval (ns since unix epoch)'),
+             ('dt', np.int16, 'Time resolution in ns'),
+             ('length', np.int32, 'Length of the interval in samples')
+            ]
+    
+    def compute(self, raw_records):
+        
+        r = raw_records[(raw_records['channel'] >= self.config['light_level'][0])&(raw_records['channel'] <= self.config['light_level'][1])]
```

Comment:
On second thought I'm not sure if we want to make a cut on the light_level (or channel_list) here anyway. Maybe it would be best to calculate for all channels at processing time and make selections like this in the analysis script? Having higher occupancy could be useful at some point maybe? It would take longer though...

Reply:
Chiara wanted to have the possibility to process the plug-in only for certain PMTs. This is way it has been added.

Reply:
The light level is set to select the channels we want to analyse for each datasets. In 1T we had 3 levels corresponding to bottom, top inner and top outer rings. So in case we want to run it over all the pmts just leave the default setting, but for the standard calibration we can just pass the list of pmts we want to run with the given light levels and reduce the time o amount of data (in 1T case by a factor 3).